src/Helldivers-2-API/Middlewares/RateLimitMiddleware.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.Extensions.Caching.Memory;
+using System.Net;
+using System.Threading.RateLimiting;
+
+namespace Helldivers.API.Middlewares;
+
+/// <summary>
+/// Handles applying rate limit logic to the API's requests.
+/// </summary>
+public sealed partial class RateLimitMiddleware(ILogger<RateLimitMiddleware> logger, IMemoryCache cache) : IMiddleware
+{
+    private const int DefaultRequestLimit = 5;
```

Comment:
These should be changed to .env variables.

We would also like to have separate rate limit policies.